app/src/main/java/org/mifos/mobile/ui/activities/BiometricAuthentication.kt

```diff
@@ -0,0 +1,106 @@
+
+import android.content.Intent
+import androidx.biometric.BiometricManager
+import android.os.Build
+import android.provider.Settings
+import android.provider.Settings.EXTRA_BIOMETRIC_AUTHENTICATORS_ALLOWED
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.biometric.BiometricPrompt
+import androidx.core.content.ContextCompat
+import androidx.core.content.ContextCompat.startActivity
+import androidx.fragment.app.FragmentActivity
+import org.mifos.mobile.R
+
+
+open class BiometricAuthentication(
+    val context: FragmentActivity,
+)  {
+    private val executor = ContextCompat.getMainExecutor(context)
+    private val callback = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+        object : BiometricPrompt.AuthenticationCallback() {
+            override fun onAuthenticationFailed() {
+                super.onAuthenticationFailed()
+            }
+
+            override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
+                super.onAuthenticationSucceeded(result)
+                val intent=Intent(context,HomeActivity::class.java)
+                context.startActivity(intent)
+            }
+
+            override fun onAuthenticationError(errorCode: Int, errString: CharSequence) {
+                super.onAuthenticationError(errorCode, errString)
+            }
+        }
+    } else {
+        TODO("VERSION.SDK_INT < P")
+    }
+
+    private val biometricPrompt = androidx.biometric.BiometricPrompt(context, executor, callback)
+
+
+
+    fun launchBiometricEnrollment(resultLauncher: ActivityResultLauncher<Intent>) {
+        val intent: Intent = when {
+            Build.VERSION.SDK_INT >= Build.VERSION_CODES.R -> {
+                Intent(Settings.ACTION_BIOMETRIC_ENROLL).putExtra(
+                    EXTRA_BIOMETRIC_AUTHENTICATORS_ALLOWED,
+                    BiometricManager.Authenticators.BIOMETRIC_STRONG or BiometricManager.Authenticators.BIOMETRIC_WEAK
+                )
+            }
+            Build.VERSION.SDK_INT >= Build.VERSION_CODES.P -> {
+                Intent(Settings.ACTION_FINGERPRINT_ENROLL)
+            }
+            else -> {
+                Intent(Settings.ACTION_SECURITY_SETTINGS)
+            }
+        }
+        resultLauncher.launch(intent)
+    }
+
+    fun getBiometricCapabilities(): BiometricCapability {
+        val biometricManager = BiometricManager.from(context)
+        return when (biometricManager.canAuthenticate(BiometricManager.Authenticators.BIOMETRIC_WEAK or BiometricManager.Authenticators.DEVICE_CREDENTIAL)) {
+            BiometricManager.BIOMETRIC_SUCCESS -> {
+                BiometricCapability.HAS_BIOMETRIC_AUTH
+            }
+            BiometricManager.BIOMETRIC_ERROR_NO_HARDWARE -> {
+                BiometricCapability.NOT_SUPPORTED
+            }
+            BiometricManager.BIOMETRIC_ERROR_HW_UNAVAILABLE -> {
+                BiometricCapability.NOT_SUPPORTED
+            }
```

Comment:
We don't need to check for conditions where the result will be `NOT_SUPPORTED` we already have it in the else block, so remove them.